### PR TITLE
feat(module:message): Add notification & message component  on close …

### PR DESCRIPTION
**Pull request by 100cm**
…callback features

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

this pr related to #2458 feature

### CODE REVIEW

**Comment by 100cm** [comment]
不是很清楚这一块的codecov 要怎么样添加测试用例了。求指导。。

File: components/message/demo/close.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+order: 2
+title:
+  zh-CN: 添加关闭message回掉函数
```
Comment (JLHwung) [inline components/message/demo/close.md]: s/回掉/回调

**Comment by wendzhue** [comment]
@vthinkxie We've discussed about adding `nzAfterClose Observable` earlier. What do you think? Callback or observable?

**Comment by 100cm** [comment]
Add `nzAfterClose` Subject to MessageService / MessageContainer  that can be subscribed anywhere ?

**Comment by wendzhue** [comment]
> Add `nzAfterClose` Subject to MessageService / MessageContainer that can be subscribed anywhere ?

Yes, like that in Modal component.

**Comment by 100cm** [comment]
I think this implementation will lead to programmer write much `service.observe.subscribe` . and need to save and judge the tmp MessageId in component

**Comment by wendzhue** [comment]
@100cm What I mean is not to subscribe the `MessageService` itself. Notice that every method creating a message or notification returns a `NzMessageDataFilled` object, on which you can add an observable like `nzAfterClose` and users can subscribe it.

**Comment by wendzhue** [comment]
@100cm Hi man. Are you still working on this?

**Comment by 100cm** [comment]
> @100cm Hi man. Are you still working on this?

i will update 2 hours later.

**Comment by wendzhue** [comment]
> > @100cm Hi man. Are you still working on this?
> 
> i will update 2 hours later.

牛逼

**Comment by JLHwung** [comment]
@wendzhue Please do not speak slangs on the thread.

**Pull request closed without merging**